src/Mvc/Mvc.Core/src/Routing/IRouteTemplateProvider.cs

```diff
@@ -3,12 +3,14 @@
 
 #nullable enable
 
+using Microsoft.AspNetCore.Routing;
+
 namespace Microsoft.AspNetCore.Mvc.Routing
 {
     /// <summary>
     /// Interface for attributes which can supply a route template for attribute routing.
     /// </summary>
-    public interface IRouteTemplateProvider
+    public interface IRouteTemplateProvider : IRoutePatternMetadata, IRouteOrderMetadata, IRouteNameMetadata
```

Comment:
It's really nice to see all this expressive MVC functionality get sprinkled into the core. 

🌈 believe in yourself 🌈